azure/scope/managedcontrolplane.go

```diff
@@ -613,6 +613,7 @@ func (s *ManagedControlPlaneScope) MakeEmptyKubeConfigSecret() corev1.Secret {
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(s.ControlPlane, infrav1.GroupVersion.WithKind("AzureManagedControlPlane")),
 			},
+			Labels: map[string]string{clusterv1.ClusterNameLabel: s.Cluster.Name},
```

Comment:
I might just be misunderstanding but is this related to the bumping CAPI to v1.5?

Reply:
Copying myself. I have the same question.

Reply:
Good catch! This does seem random, but deep in the CAPI v1.5.0 [migration guide](https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.4-to-v1.5#other) there is this comment:
> The generated kubeconfig by the Control Plane providers must be labelled with the key-value pair cluster.x-k8s.io/cluster-name=${CLUSTER_NAME}. This is required for the CAPI managers caches to store and retrieve them for the required operations.